packages/next/src/server/route-matcher-providers/dev/dev-app-route-route-matcher-provider.ts

```diff
@@ -39,6 +40,9 @@ export class DevAppRouteRouteMatcherProvider extends FileCacheRouteMatcherProvid
   ): Promise<ReadonlyArray<AppRouteRouteMatcher>> {
     const matchers: Array<AppRouteRouteMatcher> = []
     for (const filename of files) {
+      // Skip static metadata files as they are served from filesystem.
+      if (isStaticMetadataFile(filename)) continue
```

Comment:
Should keep consistent to also strip the appDir prefix (like the vercel comment above). You're testing the absolute path here. It will also return true but it's better to keep it accurate